### PR TITLE
No footnotes in read time

### PR DESCRIPTION
**Pull request by oetherington**
Bugfix/feature from pairing with Sarah.

The change going forward is simple enough, but the migration takes a _very_ long time to run (I'd estimate ~1 day on production) - it may or may not be worth it.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202933923679455) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/editor/make_editable_callbacks.ts [hunk 1]
```diff
@@ -277,10 +277,30 @@ export function dataToMarkdown(data, type) {
...
+ * the document.
+ */
 export async function dataToWordCount(data, type) {
   try {
-    const markdown = dataToMarkdown(data, type) || ""
-    return markdown.split(" ").length
+    const markdown = dataToMarkdown(data, type) ?? "";
+    const withoutFootnotes = markdown
+      .replace(/^1\. {2}\^\*\*\[\^\]\(#(.|\n)*/m, "")
+      .replace(/^\[\^\d+\]:.*/m, "");
```
Comment (s-cheng) [inline packages/lesswrong/server/editor/make_editable_callbacks.ts:301]: Why check for `1` in the first regex and `\d` in the second? I feel like we should be consistent between them. Also it seems like even in the second case, the footnote can spill into multiple lines: https://www.markdownguide.org/extended-syntax/#footnotes

And I guess it doesn't even need to use a number, you can use a string (though this case is prob rare and might be worth skipping).

File: packages/lesswrong/server/migrations/2022-08-31-recomputeWordCounts.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                      }
+                    }
+                  });
+                }
+              }
+            }
+            // eslint-disable-next-line no-console
+            console.log(`Writing ${updates.length} word counts for ${collectionName}`);
+            if (updates.length) {
+              await Revisions.rawCollection().bulkWrite(updates, { ordered: false });
```
Comment (s-cheng) [inline packages/lesswrong/server/migrations/2022-08-31-recomputeWordCounts.ts:69]: This should be `collection.rawCollection()...` right?

File: packages/lesswrong/server/migrations/2022-08-31-recomputeWordCounts.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          }
+        }
+        // eslint-disable-next-line no-console
+        console.log(`Writing ${updates.length} word counts for Revisions`);
+        if (updates.length) {
+          await Revisions.rawCollection().bulkWrite(updates, { ordered: false });
+        }
+      },
+    });
+    for (const collectionName of editableCollections) {
```
Comment (s-cheng) [inline packages/lesswrong/server/migrations/2022-08-31-recomputeWordCounts.ts:40]: You modeled this off of `2019-02-14-computeWordCounts.ts` right? Could you add more comments in this file explaining what this is doing?

Also, maybe we could just call this on the post contents field, instead of all editable fields? Is that the only field we really care about?

File: packages/lesswrong/server/editor/make_editable_callbacks.ts (hunk 1 above)
Comment (oetherington) [reply packages/lesswrong/server/editor/make_editable_callbacks.ts:301]: Good point! I've changed the second regex to check for "1" instead. I've ignored the case where they can enter strings for now since it's an edge case as you say, and I think it will make it more likely that we hit false positive matches. I've also added a couple of unit tests for this behaviour.

File: packages/lesswrong/server/migrations/2022-08-31-recomputeWordCounts.ts (hunk 3 above)
Comment (oetherington) [reply packages/lesswrong/server/migrations/2022-08-31-recomputeWordCounts.ts:40]: I just added a brief comment to the file describing how it works and why it's separate from the other migration - let me know if you think it needs any more detail.

I think you're right that we only need to update revisions and post bodies in this case. I'll leave the migration as is though because it may be a useful utility in the future, and I'll just filter out the other collections before I run it.

File: packages/lesswrong/server/editor/make_editable_callbacks.ts (hunk 1 above)
Comment (s-cheng) [reply packages/lesswrong/server/editor/make_editable_callbacks.ts]: Thanks! I think we'll also want `(.|\n)*` in the second regex as well, since that can include newlines too?

**Comment by oetherington** [comment]
Fixes #5417
